test/src/main/java/org/apache/accumulo/test/BatchWriterInTabletServerIT.java

```diff
@@ -104,24 +104,34 @@ private void test(String t1, String t2, Connector c, IteratorSetting itset, int
     c.tableOperations().attachIterator(t2, summer);
 
     Map.Entry<Key,Value> actual;
-    // Scan t1 with an iterator that writes to table t2
-    Scanner scanner = c.createScanner(t1, Authorizations.EMPTY);
-    scanner.addScanIterator(itset);
-    actual = Iterators.getOnlyElement(scanner.iterator());
-    Assert.assertTrue(actual.getKey().equals(k, PartialKey.ROW_COLFAM_COLQUAL));
-    Assert.assertEquals(BatchWriterIterator.SUCCESS_VALUE, actual.getValue());
-    scanner.close();
+    Scanner scanner = null;
+    try {
+      // Scan t1 with an iterator that writes to table t2
+      scanner = c.createScanner(t1, Authorizations.EMPTY);
+      scanner.addScanIterator(itset);
+      actual = Iterators.getOnlyElement(scanner.iterator());
+      Assert.assertTrue(actual.getKey().equals(k, PartialKey.ROW_COLFAM_COLQUAL));
+      Assert.assertEquals(BatchWriterIterator.SUCCESS_VALUE, actual.getValue());
+    } finally {
+      if (scanner != null) {
+        scanner.close();
+      }
+    }
 
-    // ensure entries correctly wrote to table t2
-    scanner = c.createScanner(t2, Authorizations.EMPTY);
-    actual = Iterators.getOnlyElement(scanner.iterator());
-    log.debug("t2 entry is " + actual.getKey().toStringNoTime() + " -> " + actual.getValue());
-    Assert.assertTrue(actual.getKey().equals(k, PartialKey.ROW_COLFAM_COLQUAL));
-    Assert.assertEquals(numEntriesToWritePerEntry, Integer.parseInt(actual.getValue().toString()));
-    scanner.close();
+    try {
+      // ensure entries correctly wrote to table t2
+      scanner = c.createScanner(t2, Authorizations.EMPTY);
+      actual = Iterators.getOnlyElement(scanner.iterator());
+      log.debug("t2 entry is " + actual.getKey().toStringNoTime() + " -> " + actual.getValue());
+      Assert.assertTrue(actual.getKey().equals(k, PartialKey.ROW_COLFAM_COLQUAL));
+      Assert.assertEquals(numEntriesToWritePerEntry, Integer.parseInt(actual.getValue().toString()));
+    } finally {
+      if (scanner != null) {
+        scanner.close();
+      }
+    }
```

Comment:
The two in this class can use try-with-resources also. It's okay that the variable will have to be declared twice instead of reused.

Reply:
I missed this comment, so my general comment, is sort of answered by this. I will do this or at least try.